flycheck.el

```diff
@@ -7735,22 +7749,20 @@ contains a cabal file."
     (`haskell-stack-ghc
      (or
       (when (buffer-file-name)
-        (locate-dominating-file (buffer-file-name) "stack.yaml"))
+        (flycheck--locate-dominating-file-matching
+         (file-name-directory
```

Comment:
It seems this was not a good idea as this is causing following issue https://github.com/flycheck/flycheck-haskell/issues/72. What happens is that `locate-dominating-file` passes file name into lambda argument, that calls `directory-files` on it and throws an error. Should we restore previous behavior of making `flycheck--locate-dominating-file-matching` accept directory names only?